packages/api-explorer/src/scenes/MethodScene/components/DocOperation.tsx

```diff
@@ -1,22 +1,90 @@
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
 import React, { FC } from 'react'
 import { IMethod } from '@looker/sdk-codegen'
-import { Text, Flex, Box } from '@looker/components'
-
+import { Badge, Code, Flex, SpaceVertical, Text } from '@looker/components'
+import { HttpMethod } from '@looker/sdk/src'
+import styled from 'styled-components'
 import { DocPseudo } from '../../../components'
 
 interface DocOperationProps {
   method: IMethod
 }
 
+function getMethodColor(method: HttpMethod) {
+  switch (method) {
+    case 'DELETE':
+      return 'critical'
+    case 'GET':
+      return 'inform'
+    case 'HEAD':
+      return 'neutral'
+    case 'PATCH':
+      return 'warn'
+    case 'POST':
+      return 'positive'
+    case 'PUT':
+      return 'key'
+    case 'TRACE':
+      return 'warn'
+  }
+}
+const StyledBadge = styled(Badge)`
+  border-radius: 4px;
+  padding-top: ${({ theme }) => theme.space.xxsmall};
+  padding-bottom: ${({ theme }) => theme.space.xxsmall};
+`
+
+const StyledEndpointLabel = styled(Text)``
+StyledEndpointLabel.defaultProps = {
+  fontSize: 'xsmall',
+  fontWeight: 'semiBold',
+  mr: 'xxsmall',
+  textTransform: 'uppercase',
+}
+
 export const DocOperation: FC<DocOperationProps> = ({ method }) => (
-  <Box mb="large">
-    <Flex>
-      <Text mr="xxsmall">SDK: </Text> <DocPseudo method={method} />
-    </Flex>
-    <Flex>
-      <Text>
-        {method.httpMethod}: {method.endpoint}
-      </Text>
-    </Flex>
-  </Box>
+  <SpaceVertical mb="xlarge" gap="xsmall">
+    <div>
+      <StyledBadge intent={getMethodColor(method.httpMethod)}>
+        <Flex alignItems="center">
+          <StyledEndpointLabel>SDK:</StyledEndpointLabel>
+          <DocPseudo method={method} />
+        </Flex>
+      </StyledBadge>
+    </div>
+    <div>
+      <StyledBadge intent={getMethodColor(method.httpMethod)}>
+        <Flex alignItems="center">
+          <StyledEndpointLabel>{method.httpMethod}:</StyledEndpointLabel>
+          <Code fontSize="small" fontWeight="normal">
+            {method.endpoint}
+          </Code>
+        </Flex>
+      </StyledBadge>
+    </div>
```

Comment:
Wrapping `div`s and`Flex` usages point at some missing components here perhaps.

`StyledBadge` could have `align-items: center:` by default to remove the flex. The outer `div` seem superflous?

Reply:
@lukelooker - This seemed to be the only way I could get it to work without the badges filling 100% width.
<img width="573" alt="Screen Shot 2020-06-24 at 12 15 34 PM" src="https://user-images.githubusercontent.com/53451500/85617622-6c40ab00-b614-11ea-9ab2-2b65f558119a.png">

Reply:
Worked with Jared to move all of this into styled `div`. Resolving this for now.